client/me/account-password/index.jsx

```diff
@@ -38,26 +41,29 @@ class AccountPassword extends React.Component {
 		password: '',
 		validation: null,
 		pendingValidation: true,
-		isUnsaved: false,
 	};
 
 	componentDidMount() {
 		this.debouncedPasswordValidate = debounce( this.validatePassword, 300 );
 	}
 
 	componentDidUpdate( prevProps ) {
-		if ( prevProps.isPendingPasswordChange && ! this.props.isPendingPasswordChange ) {
-			// eslint-disable-next-line react/no-did-update-set-state
-			this.setState( { isUnsaved: false } );
+		if (
+			prevProps.isPendingPasswordChange &&
+			! this.props.isPendingPasswordChange &&
+			! this.props.hasUserSettingsRequestFailed
+		) {
+			this.props.markSaved();
+			window.location = window.location.pathname + '?updated=password';
```

Comment:
`window.location = '?updated=password'` also works because the URL is relative. Adds/modifies the query string while keeping the path.

Reply:
https://github.com/Automattic/wp-calypso/pull/52054/commits/1275947371e36a8f7945c152b2f4592ad9c30ca6 👍🏻